pandas/tests/groupby/test_categorical.py

```diff
@@ -433,6 +433,35 @@ def test_observed_groups_with_nan(observed):
     tm.assert_dict_equal(result, expected)
 
 
+def test_observed_ops(agg_func):
+    cat = pd.Categorical(['a', np.nan, np.nan], categories=['a', 'b', 'c'])
+    ser = pd.Series([1., 2., 3.])
+    df = pd.DataFrame({'cat': cat, 'ser': ser})
+
+    grp = df.groupby('cat', observed=False)['ser']
+    func = getattr(grp, agg_func)
+
+    if agg_func == 'nth':  # Need an argument
+        result = func(0)
+    else:
+        result = func()
+
+    if agg_func == 'sum':  # TODO: maybe a bug?
```

Comment:
Seemed strange that sum didn't return NA values where as the others did. Prod definitely seems off as well - I think these are existing bugs but can check in more detail in a few days

Reply:
This looks to be consistent with one of the [changes described in 0.22.0](http://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.22.0.html#grouping-by-a-categorical).

Reply:
Great find. I guess this is intentional then just to support sum - cc @TomAugspurger if you have any insights

Reply:
I haven't looked at the context, but @jschendel is correct that we define `sum([])` to be 0 and `prod([])` to be 1.

Reply:
Good to know - I'll fix some of the comments up on next iteration